src/vs/workbench/contrib/notebook/browser/notebookExecutionStateServiceImpl.ts

```diff
@@ -187,30 +187,29 @@ export class NotebookExecutionStateService extends Disposable implements INotebo
 	}
 
 	private _getFailedCellListener(notebook: NotebookTextModel): IDisposable {
-		return this._register(
-			notebook.onWillAddRemoveCells((e: NotebookTextModelWillAddRemoveEvent) => {
-				const lastFailedCell = this._lastFailedCells.get(notebook.uri)?.cellHandle;
-				if (lastFailedCell !== undefined) {
-					const lastFailedCellPos = notebook.cells.findIndex(c => c.handle === lastFailedCell);
-					e.rawEvent.changes.forEach(([start, deleteCount, addedCells]) => {
-						if (deleteCount) {
-							if (lastFailedCellPos >= start && lastFailedCellPos < start + deleteCount) {
-								this._setLastFailedCellVisibility(notebook.uri, false);
-							}
+		return notebook.onWillAddRemoveCells((e: NotebookTextModelWillAddRemoveEvent) => {
+			const lastFailedCell = this._lastFailedCells.get(notebook.uri)?.cellHandle;
+			if (lastFailedCell !== undefined) {
+				const lastFailedCellPos = notebook.cells.findIndex(c => c.handle === lastFailedCell);
+				e.rawEvent.changes.forEach(([start, deleteCount, addedCells]) => {
+					if (deleteCount) {
+						if (lastFailedCellPos >= start && lastFailedCellPos < start + deleteCount) {
+							this._setLastFailedCellVisibility(notebook.uri, false);
 						}
+					}
 
-						if (addedCells.length > 0) {
-							addedCells.some(cell => {
-								if (cell.handle === lastFailedCell) {
-									this._setLastFailedCellVisibility(notebook.uri, true);
-									return true;
-								}
-								return false;
-							});
-						}
-					});
-				}
-			}));
+					if (addedCells.length > 0) {
+						addedCells.some(cell => {
```

Comment:
Didn't notice this before, the benefit of `some` is that it returns whether some item in the array matches the condition. I'd write it like this:

```ts
if (addedCells.some(cell => cell.handle === lastFailedCell)) {
    this._setLastFailedCellVisibility(notebook.uri, true);
}
```